### PR TITLE
[DPM-133] prng

### PR DESCRIPTION
**Pull request by cyphersnake**
Add prng and add service.hpp file

### CODE REVIEW

File: sdk/include/game-contract-sdk/game_base.hpp [hunk 1]
```diff
@@ -16,6 +15,8 @@
 #include <game-contract-sdk/rsa.hpp>
 #include <game-contract-sdk/dispatcher.hpp>
 
+#include "service.hpp"
```
Comment (algys) [inline sdk/include/game-contract-sdk/game_base.hpp]: better use '<>' notation to include

File: sdk/include/game-contract-sdk/service.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+// xoshiro256++ prng algo
+// http://prng.di.unimi.it/
+class PRNG {
+public:
+    explicit PRNG(checksum256 && seed) : _s(split(std::move(seed))) {
+    }
+
+    explicit PRNG(std::array<uint64_t, 4> && seed) : _s(std::move(seed)) {
```
Comment (algys) [inline sdk/include/game-contract-sdk/service.hpp:46]: add also `const& seed` constructor

File: sdk/include/game-contract-sdk/game_base.hpp (hunk 1 above)
Comment (cyphersnake) [reply sdk/include/game-contract-sdk/game_base.hpp]: Why? `<>` for global imports and `""` for local. If we always use `<>` - need to append it to code style (and create it)

Comment (algys) [reply sdk/include/game-contract-sdk/game_base.hpp]: Just for convenience, explicit definition of project and sub-project in include path make more simpler to reading code and also it helps to write include paths in cmake files.

Comment (algys) [reply sdk/include/game-contract-sdk/game_base.hpp]: basically, some guides says that need to define full path related to project base in quotes, but in that case `<>`  is more likely :)

File: sdk/include/game-contract-sdk/game_base.hpp [hunk 3]
```diff
@@ -16,6 +15,8 @@
 #include <game-contract-sdk/rsa.hpp>
 #include <game-contract-sdk/dispatcher.hpp>
 
+#include <service.hpp>
```
Comment (algys) [inline sdk/include/game-contract-sdk/game_base.hpp]: I suppose like `#include <game-contract-sdk/service.hpp>` ...

**Review by algys — APPROVED** [review]

**Review by Eenae — COMMENTED** [review]
approved

**Pull request merged**